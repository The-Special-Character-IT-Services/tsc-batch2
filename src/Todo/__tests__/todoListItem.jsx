import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoListItem from '../todoListItem';

const toggleComplete = jest.fn();

const deleteTodo = jest.fn();

const item = {
  text: 'asdfasdf',
  isDone: false,
  id: 2,
};

const completedItem = {
  text: 'get milk',
  isDone: true,
  id: 3,
};

const setup = data =>
  render(
    <TodoListItem
      item={data}
      toggleComplete={toggleComplete}
      deleteTodo={deleteTodo}
      isUpdating={undefined}
      isDeleting={undefined}
    />,
  );

test('should render todoListItem Component', () => {
  const { container } = setup(item);

  expect(container.firstChild).toBeDefined();
});

test('should take snapshot of the component', () => {
  const { container } = setup(item);
  expect(container.firstChild).toMatchSnapshot();
});

test('should click delete button', () => {
  const { queryByRole } = setup(item);

  const deleteBtn = queryByRole('button');

  expect(deleteBtn).not.toBeNull();

  fireEvent.click(deleteBtn);
  fireEvent.click(deleteBtn);

  expect(deleteTodo).toBeCalledTimes(2);
  expect(deleteTodo).toBeCalledWith(item.id);
});

test('should change checkbox value', () => {
  const { queryByTestId } = setup(item);

  const isDoneCheckbox = queryByTestId('isDone-checkbox');

  expect(isDoneCheckbox).not.toBeNull();

  fireEvent.click(isDoneCheckbox);

  expect(toggleComplete).toBeCalledTimes(1);
  expect(toggleComplete).toBeCalledWith(item);
});

test('should line-through not available for isdone false', () => {
  const { queryByTestId } = setup(item);
  const todoText = queryByTestId('todo-text');
  expect(todoText).not.toHaveClass('line-through');
});

test('should line-through  available for isdone true', () => {
  const { queryByTestId } = setup(completedItem);
  const todoText = queryByTestId('todo-text');
  expect(todoText).toHaveClass('line-through');
});
