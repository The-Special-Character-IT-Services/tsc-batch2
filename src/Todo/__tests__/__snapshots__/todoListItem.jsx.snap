// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take snapshot of the component 1`] = `
<div
  class="flex items-center m-4"
>
  <input
    class="disabled:text-gray-400 disabled:border-gray-200"
    data-testid="isDone-checkbox"
    type="checkbox"
  />
  <p
    class="flex-1 px-4 truncate "
    data-testid="todo-text"
  >
    asdfasdf
  </p>
  <button
    class="btn-primary disabled:bg-gray-400"
    type="button"
  >
    Delete Item
  </button>
</div>
`;
